src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -27,7 +28,7 @@ namespace NuGet.SolutionRestoreManager
     [PackageRegistration(UseManagedResourcesOnly = true, AllowsBackgroundLoading = true)]
     [ProvideAutoLoad(VSConstants.UICONTEXT.SolutionExists_string)]
     [Guid(PackageGuidString)]
-    public sealed class RestoreManagerPackage : AsyncPackage
+    public sealed class RestoreManagerPackage : AsyncPackage, IVsUpdateSolutionEvents4
```

Comment:
Discussed offline, it's required to be here since we want to Advise to these events as soon as solution loads.